src/test/java/batteryreceiver/TestBatteryPropertyReceiver.java

```diff
@@ -0,0 +1,186 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import batteryevaluator.BatteryPropertyEvaluator;
+
+public class TestBatteryPropertyReceiver {
+	BatteryPropertyEvaluator propertyEvaluator;
+	List<Float> socList;
+	List<Float> emptySocList;
+	List<Float> nullSocList;
+	List<Float> socListWith3Items;
+	
+	List<Float> chargeRateList;
+	List<Float> emptyChargeRateList;
+	List<Float> nullChargeRateList;
+	List<Float> chargeRateListWith3Items;
+	
+ 	@Before
+	public void setup() {
+ 		propertyEvaluator = new BatteryPropertyEvaluator();
+ 		
+ 		socList = new ArrayList<Float>();
+ 		socList.add(78.66838f);
+ 		socList.add(52.88753f);
+ 		socList.add(44.716034f);
+ 		socList.add(74.336235f);
+ 		socList.add(76.90123f);
+ 		
+ 		emptySocList = new ArrayList<Float>();
+ 		nullSocList = null;
+ 		
+ 		socListWith3Items = new ArrayList<Float>();
+ 		socListWith3Items.add(78.66838f);
+ 		socListWith3Items.add(52.88753f);
+ 		socListWith3Items.add(44.716034f);
+ 		
+ 		
+ 		chargeRateList = new ArrayList<Float>();
+ 		chargeRateList.add(0.12779322f);
+ 		chargeRateList.add(0.572595f);
+ 		chargeRateList.add(0.65613645f);
+ 		chargeRateList.add(0.024554348f);
+ 		chargeRateList.add(0.72287005f);
+ 		
+ 		emptyChargeRateList = new ArrayList<Float>();
+ 		nullChargeRateList = null;
+ 		
+ 		chargeRateListWith3Items = new ArrayList<Float>();
+ 		chargeRateListWith3Items.add(0.12779322f);
+ 		chargeRateListWith3Items.add(0.572595f);
+ 		chargeRateListWith3Items.add(0.65613645f);
+ 		
+	}
+	
+	
+	@Test
+	public void givenSocListisNull_whenEvaluatingMaxOfSoc_thenReturnZero() {
+		Float maxSoc = propertyEvaluator.getMaxSocValue(nullSocList);
+		assertEquals(Float.valueOf(0), maxSoc);
+	}
+	
+	@Test
+	public void givenSocListisNull_whenEvaluatingMinOfSoc_thenReturnZero() {
+		Float minSoc = propertyEvaluator.getMinSocValue(nullSocList);
+		assertEquals(Float.valueOf(0), minSoc);
+	}
+	
+	@Test
+	public void givenSocListisNull_whenEvaluatingAvgOfSoc_thenReturnZero() {
+		Double averageSoc = propertyEvaluator.getAverageOfSocList(nullSocList);
+		assertEquals(Double.valueOf(0), averageSoc);
+	}
+	
+	@Test
+	public void givenSocListisEmpty_whenEvaluatingMaxOfSoc_thenReturnZero() {
+		Float maxSoc = propertyEvaluator.getMaxSocValue(emptySocList);
+		assertEquals(Float.valueOf(0), maxSoc);
+	}
+	
+	@Test
+	public void givenSocListisEmpty_whenEvaluatingMinOfSoc_thenReturnZero() {
+		Float minSoc = propertyEvaluator.getMinSocValue(emptySocList);
+		assertEquals(Float.valueOf(0), minSoc);
+	}
+	
+	@Test
+	public void givenSocListisEmpty_whenEvaluatingAvgOfSoc_thenReturnZero() {
+		Double averageSoc = propertyEvaluator.getAverageOfSocList(emptySocList);
+		assertEquals(Double.valueOf(0), averageSoc);
+	}
+	
+	@Test
+	public void givenSocList_whenEvaluatingMaxOfSoc_thenReturnMax() {
+		Float maxSoc = propertyEvaluator.getMaxSocValue(socList);
+		assertEquals(Float.valueOf(78.66838f), maxSoc);
```

Comment:
Use a tolerance epsilon when comparing floating-point for equality. Consider what accuracy will be required for your application. Writing tests will bring out such assumptions.